File: Python/SummaryRanges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        myL = []
        i=0
        while len(nums)>i:
            start = i
            while len(nums)>i+1 and nums[i+1]==nums[i]+1:
                i=i+1
            if start == i:
                myL.append(str(nums[start]))
            else :
                myL.append(str(nums[start])+"->"+str(nums[i]))
            i=i+1
        return myL

File: Vectors/SummaryRanges.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> result;
        int i=0;
        sz = nums.size();
        for(int i=0;i<sz;i++)
        {
            int start = i;
            while(sz>i+1 && nums[i+1]==nums[i]+1) i++;
            if(start ==i) result.push_back(to_string(nums[start]));
            else result.push_back(to_string(nums[start])+"->"+to_string(nums[i]));
        }
        return result;
    }
private:
    int sz;
};